perun-core/src/main/java/cz/metacentrum/perun/core/entry/MembersManagerEntry.java

```diff
@@ -1341,6 +1341,32 @@ public RichMember sponsorMember(PerunSession session, Member sponsored, User spo
 		return membersManagerBl.convertMembersToRichMembers(sess, membersManagerBl.getSponsoredMembers(sess, vo));
 	}
 
+	@Override
+	public List<MemberWithSponsors> getSponsoredMembersAndTheirSponsors(PerunSession sess, Vo vo, List<String> attrNames) throws VoNotExistsException, PrivilegeException, AttributeNotExistsException {
+		Utils.checkPerunSession(sess);
+		Utils.notNull(vo, "vo");
+
+		perunBl.getVosManagerBl().checkVoExists(sess, vo);
+
+		//Authorization
+		if(!AuthzResolver.authorizedInternal(sess, "getSponsoredMembersAndTheirSponsors_Vo_policy", vo)) {
+			throw new PrivilegeException(sess, "getSponsoredMembersAndTheirSponsors");
+		}
+
+		List<AttributeDefinition> attrsDef = new ArrayList<>();
+		for (String attrName : attrNames) {
+			attrsDef.add(getPerunBl().getAttributesManagerBl().getAttributeDefinition(sess, attrName));
+		}
+
+		List<RichMember> richMembers = membersManagerBl.convertMembersToRichMembersWithAttributes(sess, getSponsoredMembers(sess, vo), attrsDef);
+		richMembers = membersManagerBl.filterOnlyAllowedAttributes(sess, richMembers, null, true);
```

Comment:
Is there any reason why do you call this version of the method instead of this one?
```suggestion
		richMembers = membersManagerBl.filterOnlyAllowedAttributes(sess, richMembers);
```

Reply:
@zlamalp why did you suggested this version of the method?

Reply:
Well, there is no reason to resolve attribute authorization for each RichMember, if you can be sure, they are from the same VO and with the same set of attributes. Hence you can use "context=true" and authorization is resolved only once and results are reused for the next member.

Reply:
Ok, I was just confused because of the group parameter.